src/hotspot/share/code/codeCache.hpp

```diff
@@ -191,6 +191,7 @@ class CodeCache : AllStatic {
   static void print_trace(const char* event, CodeBlob* cb, int size = 0) PRODUCT_RETURN;
   static void print_summary(outputStream* st, bool detailed = true); // Prints a summary of the code cache usage
   static void log_state(outputStream* st);
+  static void write_perf_map(outputStream* st);
```

Comment:
Seems weird for this function to have an outputStream parameter only to write the dump to an unrelated file and ignore the stream for everything but the final message.

I would either pass in the file name as an option - preferably configurable - and write the last message out here; or just write the whole perf dump to the outputstream itself, piping it to jcmd and let the caller do what he wants with it (e.g. just redirecting). The latter is what most commands do. Not sure how large the perf dump gets though, may be impractical.

Reply:
OK. I think I'll change it so `write_perf_map()` writes to the `outputStream` and then `PerfMapDCmd::execute()` handles redirecting it to a file. I don't think it makes sense to write it directly to the jcmd output though.